### DISCUSSION TITLE
IRAF mask images with sigma clipping

### DISCUSSION THREAD

**Discussion by Firestar-Reimu** [iraf-community/iraf#399] [Q&A]
I have a school homework which **MUST** use the #$%^&* IRAF

First I have to produce a median background from this image

[![enter image description here][1]][1]

but before `fmedian image x y` I need to mask out all the stars

I am going to do this from a sigma clipping, for example mask out all pixels with sigma>3

In photutils I can do:

```
sigma_clip = SigmaClip(sigma=3.0, maxiters=10)
threshold = detect_threshold(data, nsigma=2.0, sigma_clip=sigma_clip)
segment_img = detect_sources(data, threshold, npixels=10)
footprint = circular_footprint(radius=10)
mask = segment_img.make_source_mask(footprint=footprint)
mean, median, std = sigma_clipped_stats(data, sigma=3.0, mask=mask)
```

But I cannot find any tasks in IRAF which can do sigma-clipping

**How to mask images with sigma clipping in IRAF?**

  [1]: https://i.sstatic.net/6ZvbfeBM.jpg

**Comment by olebole** [comment]
The trick may be to use the "search" function of [the documentation](https://iraf.readthedocs.io). Or to call a Python script from IRAF.

**Reply by Firestar-Reimu** [reply]
I have searched the documentation for hundreds of times ... (sad)

**Comment by Firestar-Reimu** [comment]
I tried `objmask` and produced a Perl file `mask.pl`, but I find that when I use `fmedian`, it cannot just ignore the masked pixels, instead it treat them as pixels with very large/small values.
